src/core/ezcurl_core.ml

```diff
@@ -353,6 +396,169 @@ let mk_res (self : t) headers body : (_ response, _) result =
     Ok { headers; code; body; info }
   with Parse_error (e, msg) -> Error (e, Curl.strerror e ^ ": " ^ msg)
 
+let sse_frame_with_event sse_f v =
+  {
+    event = Some v;
+    id = !sse_f.id;
+    data = !sse_f.data;
+    retry = !sse_f.retry;
+    empties = !sse_f.empties;
+  }
+
+let sse_frame_with_id sse_f v =
+  {
+    event = !sse_f.event;
+    id = Some v;
+    data = !sse_f.data;
+    retry = !sse_f.retry;
+    empties = !sse_f.empties;
+  }
+
+let sse_frame_append_data sse_f v =
+  let data =
+    match !sse_f.data with
+    | None -> Some v
+    | Some vv -> Some (vv ^ "\n" ^ v)
+  in
+  {
+    event = !sse_f.event;
+    id = !sse_f.id;
+    data;
+    retry = !sse_f.retry;
+    empties = !sse_f.empties;
+  }
+
+let sse_frame_with_retry sse_f v =
+  let retry = int_of_string_opt v in
+  {
+    event = !sse_f.event;
+    id = !sse_f.id;
+    data = !sse_f.data;
+    retry;
+    empties = !sse_f.empties;
+  }
+
+let sse_frame_append_empties sse_f v =
+  let empties =
+    match !sse_f.empties with
+    | [] -> [ v ]
+    | vv -> v :: vv
+  in
+  {
+    event = !sse_f.event;
+    id = !sse_f.id;
+    data = !sse_f.data;
+    retry = !sse_f.retry;
+    empties;
+  }
+
+let sse_process_pair k v sse_f =
+  match k with
+  | "event" -> sse_f := sse_frame_with_event sse_f v
+  | "id" -> sse_f := sse_frame_with_id sse_f v
+  | "data" -> sse_f := sse_frame_append_data sse_f v
+  | "retry" -> sse_f := sse_frame_with_retry sse_f v
+  | "" -> () (* The field is ignored *)
+  | _ -> sse_f := sse_frame_append_empties sse_f k
+
+let sse_split_line s =
+  let l = String.length s in
+  let sq = String.to_seq s in
+  match Seq.find_index (fun c -> c = ':') sq with
```

Comment:
there really is no need to bring Seq in here at all :). `List.exists` will be faster and simpler, or even in this case `List.mem ':' s` is enough.

edit: even better here, sorry, is `String.index`! Way faster.